### PR TITLE
T67: complete a documented deprecation in the database connection contract

### PR DESCRIPTION
**Pull request by tomj12k**
Tracked as **T67** in the private `Morelitea/security` repository.

The finding, reasoning, verification, and review notes live there. This description is deliberately minimal because this repository is public.

**Scope:** the database startup contract, typed configuration, operator documentation, and boundary tests.

**Verified:** the amended exact head passes the focused database/startup suites and repository lint, format, type, and diff gates. Reviewers should use T67 / WI-38 in `Morelitea/security` for detail.

### CODE REVIEW

**Comment by tomj12k** [comment]
Pushed fixes for every finding in the automated review. Each one was right.

**Comment by tomj12k** [comment]
T67 exact-head follow-up is in `dc4994254`.

The boundary cases were observed red before implementation and green afterward. Final local evidence: 104 changed-module tests, 16 startup/caller tests, repository-wide lint and format, type checking, and diff validation all pass.

The finding, operational trade-off, and uncertainty analysis are recorded under private T67 / WI-38 in `Morelitea/security`. The separate image-gate status is addressed by [#1594](https://github.com/Morelitea/initiative/pull/1594).

**Comment by tomj12k** [comment]
Final hosted status at `dc4994254`: every branch-specific application gate passes, including the 30-minute backend suite. The sole remaining image-gate failure is addressed by the masked T60 follow-up [#1594](https://github.com/Morelitea/initiative/pull/1594), where the same image scan passes; merge or rebase it before treating this branch as fully green.

File: backend/app/core/config.py [hunk 1]
```diff
@@ -167,6 +167,14 @@ class Settings(BaseSettings):
...
     # them; a deployment that provisions its database out of band never sets it.
     DATABASE_URL_BOOTSTRAP: str | None = None
+    # An escape hatch, not a supported configuration. ``DATABASE_URL`` holding
+    # SUPERUSER or BYPASSRLS voids every row-level-security guarantee in
+    # SECURITY.md -- the tenancy boundary is enforced by RLS, and these two
+    # attributes are exactly the right to ignore it. Startup refuses such a
+    # connection. An operator who cannot migrate in the same maintenance window
+    # can set this to keep booting; it is recorded at WARNING on every boot so
+    # it cannot become the quiet steady state.
+    ALLOW_PRIVILEGED_DATABASE_URL: bool = False
```
[the remark this answers is not in the document]
Comment (tomj12k) [reply backend/app/core/config.py]: Considered and declining this one, which makes it the odd one out — I took the same finding on three sibling branches and acted on two. Reasoning, so it is easy to overrule.

CLAUDE.md draws the line at comments that hand a reader a method rather than a meaning, and gives the reason: this source is public.

What is actually written here does not do that:

- **The privilege names are the subject of the check, not a disclosure.** The function is `reject_privileged_database_url`; documenting it without naming what it refuses leaves a reader unable to tell a correct configuration from a refused one. That is the functional explanation the rule asks for.
- **Neither name is a repository secret.** Both are standard Postgres role attributes with published semantics. And this repository's own public `SECURITY.md` already sets out the tenancy model, the per-guild roles, `WITH INHERIT FALSE` and `FORCE ROW LEVEL SECURITY` in more detail than any comment here.
- **There is no method described.** No narrative, no sequence, no "someone could". A reader who already holds that level of database access does not need a comment to tell them what it means.

The three I did change were different in kind: each named a method or an adversary rather than a property of the code. This one names a role attribute and the setting that governs it.

**The tightening I would accept if you disagree** is one clause — the WARNING log tells the operator plainly that the documented access rules are not in force for that connection. I left it because it is a runtime message to someone who set `ALLOW_PRIVILEGED_DATABASE_UNTIL` deliberately, and its bluntness is the point: it is the only thing standing between a temporary escape hatch and a permanent one. Say the word and I will soften it.

Resolving so the list reflects that it was looked at rather than skipped. Reopen if you read the line differently — you own the standard.

— Claude Opus 5 (1M context), review at head `e41f0ee`.
